modules/post/windows/manage/reflective_dll_inject.rb

```diff
@@ -6,45 +6,167 @@
-      'Description'   => %q{
-        This module will inject into the memory of a process a specified Reflective DLL.
-      },
-      'License'      => MSF_LICENSE,
-      'Author'       => [ 'Ben Campbell'],
-      'Platform'     => [ 'win' ],
-      'SessionTypes' => [ 'meterpreter' ],
-      'References'   =>
-        [
-          [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
-        ]
-    ))
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::ReflectiveDLLInjection
 
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name' => 'Windows Manage Reflective DLL Injection
+                      Module with arguments',
+                      'Description' => '
+                      This module will inject a specified reflective DLL
+                      into the memory of a process, new or existing,
+                      passing any arguments.
+                      ',
+                      'License' => MSF_LICENSE,
+                      'Author' => 'Ben Campbell,b4rtik',
+                      'Platform' => 'win',
+                      'SessionTypes' => ['meterpreter'],
+                      'References'   =>
+                        [
+                          [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
+                        ]))
     register_options(
       [
-        OptPath.new('PATH',[true, 'Reflective DLL to inject into memory of a process.']),
-        OptInt.new('PID',[true, 'Process Identifier to inject of process to inject payload.']),
-      ])
+        OptPath.new('PATH', [true, 'Reflective DLL to inject into memory of a process.']),
+        OptInt.new('PID', [false, 'Pid  to inject', 0]),
+        OptString.new('PROCESS', [false, 'Process to spawn','notepad.exe']),
+        OptString.new('ARGUMENTS', [false, 'Command line arguments']),
+        OptInt.new('WAIT', [false, 'Time in seconds to wait.', 0])
+      ], self.class
+    )
   end
 
-  # Run Method for when run command is issued
   def run
-    # syinfo is only on meterpreter sessions
+    dll_path = gen_dll_path
+    if File.file?(dll_path)
+      client.response_timeout=20
+      run_dll(dll_path)
+    else
+      print_bad("Dll not found #{dll_path}")
+    end
+  end
+
+  def gen_dll_path
+    dll_path = ::File.expand_path(datastore['PATH'])
+    dll_path
+  end
+
+  def sanitize_process_name(process_name)
+    out_process_name = if process_name.split(//).last(4).join.eql? '.exe'
+                         process_name
+                       else
+                         process_name + '.exe'
+                       end
+    out_process_name
+  end
+
+  def pid_exists(pid)
+    return true
+  end
+
+  def launch_process
+    process_name = sanitize_process_name(datastore['PROCESS'])
+    print_status("Launching #{process_name} to inject Dll...")
+    channelized = if datastore['WAIT'] != 0
```

Comment:
This variable is defined but not used.

If you do use it, you could short hand this to simply

```suggestion
channelized = datastore['WAIT'] != 0
```

Reply:
In this case, the variable is necessary and should be used in L82